addon/components/array-container.js

```diff
@@ -131,29 +159,71 @@ export default Component.extend(HookMixin, PropTypeMixin, {
   },
 
   @readOnly
-  @computed('bunsenId', 'readOnly', 'value')
-  items (bunsenId, readOnly, value) {
+  @computed('bunsenModel')
+  itemsModel (model) {
+    const itemModels = model.items
+    if (Array.isArray(itemModels) && model.tuple) {
+      return model.additionalItems
+    }
+  },
+  @readOnly
+  @computed('bunsenId', 'readOnly', 'value', 'bunsenModel', 'cellConfig.arrayOptions.itemCell')
+  /* eslint-disable complexity*/
+  items (bunsenId, readOnly, value, bunsenModel, cellConfig) {
+    const itemModels = bunsenModel.items
     if (typeOf(value) === 'object' && 'asMutable' in value) {
       value = value.asMutable({deep: true})
     }
 
     const items = get(value || {}, bunsenId) || []
-
+    const getModel = Array.isArray(bunsenModel.items)
+      ? (item, index) => bunsenModel.items[index] || bunsenModel.additionalItems
+      : () => bunsenModel.items
+    const getCellConfig = Array.isArray(cellConfig)
+      ? (item, index) => cellConfig[index] || cellConfig[cellConfig.length - 1] || {}
+      : () => cellConfig || {}
```

Comment:
Yeah this really makes me wish we had conditional expressions like they do in some other languages (*cough* Elm *cough*)